test/adapters/optimisticAuctionRebalanceExtensionV1.spec.ts

```diff
@@ -0,0 +1,729 @@
+import "module-alias/register";
+
+import { Address, Account } from "@utils/types";
+import { ADDRESS_ZERO, ZERO } from "@utils/constants";
+import {
+  BaseManager,
+  ConstantPriceAdapter,
+  OptimisticAuctionRebalanceExtensionV1,
+  OptimisticOracleV3Mock,
+  StandardTokenMock,
+} from "@utils/contracts/index";
+import { SetToken } from "@utils/contracts/setV2";
+import DeployHelper from "@utils/deploys";
+import {
+  addSnapshotBeforeRestoreAfterEach,
+  ether,
+  getAccounts,
+  getSetFixture,
+  getWaffleExpect,
+  bitcoin,
+  usdc,
+  getTransactionTimestamp,
+  getRandomAccount,
+} from "@utils/index";
+import { SetFixture } from "@utils/fixtures";
+import { BigNumber, ContractTransaction, utils } from "ethers";
+import base58 from "bs58";
+
+const expect = getWaffleExpect();
+
+function bufferToHex(buffer: Uint8Array) {
+  let hexStr = "";
+
+  for (let i = 0; i < buffer.length; i++) {
+    const hex = (buffer[i] & 0xff).toString(16);
+    hexStr += hex.length === 1 ? "0" + hex : hex;
+  }
+
+  return hexStr;
+}
+
+// Base58 decoding function (make sure you have a proper Base58 decoding function)
+function base58ToHexString(base58String: string) {
+  const bytes = base58.decode(base58String); // Decode base58 to a buffer
+  const hexString = bufferToHex(bytes.slice(2)); // Convert buffer to hex, excluding the first 2 bytes
+  return "0x" + hexString;
+}
```

Comment:
let's move this to a utility since we also define it in global extension tests

https://github.com/IndexCoop/index-coop-smart-contracts/blob/ba40dae30c5f1b556f0cfbdcc5ce067c3cca3e25/test/global-extensions/globalOptimisticAuctionRebalanceExtension.spec.ts#L28-L44

Reply:
added here [e2d2340](https://github.com/IndexCoop/index-coop-smart-contracts/pull/157/commits/e2d2340da0c518e2d21c7e58ed3f33b7fb5c48de)